aspnetcore/security/key-vault-configuration.md

```diff
@@ -0,0 +1,144 @@
+    * Hierarchical values (configuration sections) use `--` (two dashes) as a separator.
+    * For the sample application, create two *Manual* secrets with the following name-value pairs:
+      * `MySecret`: `secret_value_1`
+      * `Section--MySecret`: `secret_value_2`
+  * Register the sample application with Azure Active Directory.
+  * Authorize the application to access the key vault. When you use the `Set-AzureRmKeyVaultAccessPolicy` PowerShell cmdlet to authorize the application to access the key vault, provide `List` and `Get` access to secrets with `-PermissionsToKeys list,get`.
+2. Update the application's *appsettings.json* file with the values of `Vault`, `ClientId`, and `ClientSecret`.
+3. Run the sample application, which obtains its configuration values from `IConfigurationRoot` with the same name as the secret name.
+  * Non-hierarchical Values: The value for `MySecret` is obtained with `config["MySecret"]`.
+  * Hierarchical Values (sections): Use `:` (colon) notation or the `.GetSection()` method.
+    * `config["Section:MySecret"]`
+    * `config.GetSection("Section")["MySecret"]`
+
+![Browser window showing secret values loaded via the Azure Key Vault Configuration Provider](key-vault-configuration/_static/sample-output.png)
+
+## Reloading secrets
+Secrets are cached until `IConfigurationRoot.Reload()` is called. Expired, disabled, and updated secrets in the key vault are not respected by the application until `Reload()` is executed.
+
+```csharp
+Configuration.Reload();
+```
+
+## Disabled and expired secrets
+Disabled and expired secrets throw a `KeyVaultClientException` at runtime. Therefore, it's important that you update your application or replace or update a disabled or expried secret in the key vault before the provider attempts to load or reload a disabled or expired secret.
+
+## Troubleshooting
+When the application fails to load configuration using the provider, an error message will be written to the [ASP.NET Logging infrastructure](xref:fundamentals/logging). The following general conditions can prevent configuration from loading:
+* The application has not been configured correctly in Azure Active Directory.
+* The key vault doesn't exist in Azure Key Vault.
+* The application has not been authorized to access the key vault.
+* The access policy doesn't include `Get` and `List` permissions.
+* In the key vault, the configuration data (name-value pair) is incorrectly named, missing, disabled, or expired.
+* The application has the wrong key vault name (`Vault`), Azure AD App Id (`ClientId`), or Azure AD Key (`ClientSecret`).
+* The Azure AD Key (`ClientSecret`) is expired.
+* The configuration key (name) is incorrect in the application for the value you're trying to load.
+
```

Comment:
Let's pretend that I know what I'm doing ...

Nobody will know the truth but us ... and [@]davidfowl. :grin: